File: src/main/java/ukalus/path/astar/Node.java
/*
 * Created on Feb 10, 2005
 *
 */
package ukalus.path.astar;

import java.util.Collection;

/**
 * @author gremopm
 *  
 */
public interface Node {

  Node getParent();

  Collection<Node> getSuccessors();

  void setCost(double cost);

  double getCost();

  void setEstimate(double estimage);

  double getEstimate();

  double getTotal();
}

File: src/main/java/ukalus/path/astar/Node.kt
/*
 * Created on Feb 10, 2005
 *
 */
package ukalus.path.astar

/**
 * @author gremopm
 */
interface Node {

    val parent: Node

    val successors: Collection<Node>

    var cost: Double

    var estimate: Double

    val total: Double
}